mixite.core/core/src/main/kotlin/org/hexworks/mixite/core/internal/impl/HexagonalGridImpl.kt

```diff
@@ -105,36 +105,38 @@ class HexagonalGridImpl<T : SatelliteData>(builder: HexagonalGridBuilder<T>) : H
         return this.hexagonDataStorage.containsCoordinate(coordinate)
     }
 
-    override fun getByCubeCoordinate(coordinate: CubeCoordinate): Maybe<Hexagon<T>> {
-        return if (containsCubeCoordinate(coordinate))
-            Maybe.of(HexagonImpl(gridData, coordinate, hexagonDataStorage))
-        else
-            Maybe.empty()
-    }
+    private fun _getByCubeCoordinate(coordinate: CubeCoordinate) =
```

Comment:
Why do you need this private function?

Reply:
Per chat, these are to avoid repeatedly creating `Maybe` instances for internal calls that either cannot be null or are adequately handled by Kotlin nullable facilities. See usage in `nearestHexagonToPoint`. I have not yet managed to add perf tests to _prove_ there is benefit to this but it would seem clear. 
Happy to rename but I think you need to decide if you want to enforce the use of `Maybe` internally to this class or not.